### PR TITLE
Add team API calls

### PR DESCRIPTION
**Pull request by RLRabinowitz**
### Issue & Steps to Reproduce / Feature Request
Feature request: Support Teams in the provider.

For starters, we need to implement the team API calls relevant for the team resource and data

### Solution
Implement get/create/update/delete/get-all API calls

### CODE REVIEW

File: examples/provider/provider.tf [hunk 1]
```diff
@@ -1,4 +1,4 @@
 provider "env0" {
-    api_key    = var.env0_api_key    # or use ENV0_API_KEY
-    api_secret = var.env0_api_secret # or use ENV0_API_SECRET
+  api_key    = var.env0_api_key    # or use ENV0_API_KEY
+  api_secret = var.env0_api_secret # or use ENV0_API_SECRET
 }
```
Comment (RLRabinowitz) [inline examples/provider/provider.tf:4]: `go fmt`

File: examples/resources/env0_cloud_credentials_project_assignment/resource.tf [hunk 2]
```diff
@@ -10,5 +10,5 @@ data "env0_project" "project" {
 
 resource "env0_cloud_credentials_project_assignment" "example" {
   credential_id = env0_aws_credentials.credentials.id
-  project_id = data.env0_project.project.id
+  project_id    = data.env0_project.project.id
```
Comment (RLRabinowitz) [inline examples/resources/env0_cloud_credentials_project_assignment/resource.tf:13]: `go fmt`

File: examples/resources/env0_configuration_variable/resource.tf [hunk 3]
```diff
@@ -6,7 +6,7 @@ resource "env0_configuration_variable" "example" {
 resource "env0_configuration_variable" "drop_down" {
   name  = "ENVIRONMENT_VARIABLE_DROP_DOWN"
   value = "first option"
-  enum  = [
+  enum = [
```
Comment (RLRabinowitz) [inline examples/resources/env0_configuration_variable/resource.tf:9]: `go fmt`

File: client/team.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
+package client
+
+import "errors"
+
+func (self *ApiClient) TeamCreate(payload TeamCreatePayload) (Team, error) {
+	if payload.Name == "" {
+		return Team{}, errors.New("Must specify team name on creation")
+	}
+	// AREL TODO: Check this
+	if payload.OrganizationId != "" {
```
Comment (liranfarage89) [inline client/team.go:10]: Done?

File: client/team_test.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+	Describe("TeamCreate", func() {
+		var createdTeam Team
+		var err error
+
+		BeforeEach(func() {
+			mockOrganizationIdCall(organizationId)
+
+			createTeamPayload := TeamCreatePayload{}
+			copier.Copy(&createTeamPayload, &mockTeam)
```
Comment (liranfarage89) [inline client/team_test.go:73]: Why creating a copy? isn't it immutable?

File: client/team.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	err = self.http.Post("/teams", payload, &result)
+	if err != nil {
+		return Team{}, err
+	}
+	return result, nil
+}
+
+func (self *ApiClient) Team(id string) (Team, error) {
+	var result Team
+	err := self.http.Get("/teams/"+id, nil, &result)
```
Comment (liranfarage89) [inline client/team.go:29]: a little out of scope, but why there is a diff between get-team to get-teams in our API? is there a special reason that you know? 
In general I expected to see /team?id={id}&orgId{org_id}... where and empty id triggers get-all

File: client/team_test.go [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+		})
+
+		It("Should return team", func() {
+			Expect(returnedTeam).To(Equal(mockTeam))
+		})
+	})
+
+	Describe("Teams", func() {
+		var returnedTeams []Team
+		mockTeams := []Team{mockTeam}
```
Comment (liranfarage89) [inline client/team_test.go:40]: Do we want to add validation tests? to those `not_empty` conditions..

File: client/team.go (hunk 6 above)
Comment (RLRabinowitz) [reply client/team.go:29]: It's a design choice we've made. I think it's technically more RESTful but I'm not sure.
One difference, for example, is that the by-id API returns a single team (not an array), and returns 404 if there's no such team

File: client/team_test.go (hunk 7 above)
Comment (RLRabinowitz) [reply client/team_test.go:41]: 1865bbd25a2fb70df22d918c174669107c722013

File: client/team_test.go (hunk 5 above)
Comment (RLRabinowitz) [reply client/team_test.go]: 1865bbd25a2fb70df22d918c174669107c722013

**Comment by HeverFarber** [comment]
@RLRabinowitz do you want to add integration tests also?

**Review by liranfarage89 — APPROVED** [review]
Nice work 👍 . maybe ask for another 👀 , since that it's the first time I'm reviewing `go` in general and our provider in particular.

File: client/team.go (hunk 4 above)
Comment (RLRabinowitz) [reply client/team.go]: Whoops

File: client/team_test.go [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Teams Client", func() {
+	mockTeam := Team{
+		Id:   "team-id",
+		Name: "team-name",
+	}
+
+	Describe("Team", func() {
```
Comment (eranelbaz) [inline client/team_test.go:18]: Get Team

File: client/team_test.go [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+		It("Should send GET request", func() {
+			httpCall.Times(1)
+		})
+
+		It("Should return team", func() {
+			Expect(returnedTeam).To(Equal(mockTeam))
+		})
+	})
+
+	Describe("Teams", func() {
```
Comment (eranelbaz) [inline client/team_test.go:39]: Get All Teams

**Review by eranelbaz — APPROVED** [review]

File: client/team_test.go (hunk 8 above)
Comment (RLRabinowitz) [reply client/team_test.go]: 4f904cdc5cbd32e466b175150d7981499f2cb8c0

File: client/team_test.go (hunk 9 above)
Comment (RLRabinowitz) [reply client/team_test.go]: 4f904cdc5cbd32e466b175150d7981499f2cb8c0

**Comment by RLRabinowitz** [comment]
[TF provider - Support team resource](https://trello.com/c/MxQpNSEQ/4032-tf-provider-support-team-resource)

**Pull request merged**